pkg/webhooks/generation.go

```diff
@@ -98,22 +99,135 @@ func (ws *WebhookServer) handleUpdate(request *v1beta1.AdmissionRequest) {
+	enqueueBool := false
+	newRes, err := enginutils.ConvertToUnstructured(request.Object.Raw)
+	if err != nil {
+		logger.Error(err, "failed to convert object resource to unstructured format")
+	}
+
+	policyName := resLabels["policy.kyverno.io/policy-name"]
+	targetSourceName := newRes.GetName()
+	targetSourceKind := newRes.GetKind()
+
+	for _, policy := range policies {
+		if policy.GetName() == policyName {
+			for _, rule := range policy.Spec.Rules {
+				if rule.Generation.Kind == targetSourceKind && rule.Generation.Name == targetSourceName {
+					data := rule.Generation.DeepCopy().Data
+					if data != nil {
+						if _, err := validate.ValidateResourceWithPattern(logger, newRes.Object, data); err != nil {
+							enqueueBool = true
+							break
+						}
+					}
+
+					cloneName := rule.Generation.Clone.Name
+					if cloneName != "" {
+						obj, err := ws.client.GetResource("", rule.Generation.Kind, rule.Generation.Clone.Namespace, rule.Generation.Clone.Name)
+						if err != nil {
+							logger.Error(err, fmt.Sprintf("source resource %s/%s/%s not found.", rule.Generation.Kind, rule.Generation.Clone.Namespace, rule.Generation.Clone.Name))
+							continue
+						}
+
+						sourceObj, newResObj := stripNonPolicyFields(obj.Object, newRes.Object, logger)
+
+						if _, err := validate.ValidateResourceWithPattern(logger, newResObj, sourceObj); err != nil {
+							enqueueBool = true
+							break
+						}
+					}
+				}
+			}
+		}
+	}
 
-			grList, err := ws.grLister.List(selector)
-			if err != nil {
-				logger.Error(err, "failed to get generate request for the resource", "label", "generate.kyverno.io/policy-name")
+	if enqueueBool {
+		grName := resLabels["policy.kyverno.io/gr-name"]
+		gr, err := ws.grLister.Get(grName)
+		if err != nil {
+			logger.Error(err, "failed to get generate request", "name", grName)
+		}
+		ws.grController.EnqueueGenerateRequestFromWebhook(gr)
+	}
+}
 
+//stripNonPolicyFields - remove feilds which get updated with each request by kyverno and are non policy fields
+func stripNonPolicyFields(obj, newRes map[string]interface{}, logger logr.Logger) (map[string]interface{}, map[string]interface{}) {
+
+	delete(obj["metadata"].(map[string]interface{})["annotations"].(map[string]interface{}), "kubectl.kubernetes.io/last-applied-configuration")
+	delete(obj["metadata"].(map[string]interface{}), "creationTimestamp")
+	delete(obj["metadata"].(map[string]interface{})["labels"].(map[string]interface{}), "generate.kyverno.io/clone-policy-name")
+	delete(obj["metadata"].(map[string]interface{}), "managedFields")
+	delete(obj["metadata"].(map[string]interface{}), "resourceVersion")
+	delete(obj["metadata"].(map[string]interface{}), "selfLink")
+	delete(obj["metadata"].(map[string]interface{}), "uid")
+	delete(obj["metadata"].(map[string]interface{}), "namespace")
+	delete(obj["metadata"].(map[string]interface{}), "name")
+	delete(newRes["metadata"].(map[string]interface{}), "managedFields")
```

Comment:
Can we extract the fields that we care about rather than delete other fields from metadata?

Reply:
We need to check kind and then extract data accordingly.

Which fields need to be extracted?

Reply:
`metadata` is the same across all resource, we just need to extract `annotations` and `labels`.

Reply:
what about fields like `type`, `spec`, `data` etc...? If any changes are made in the mentioned fields, this changes should also be considered as these fields are also dependent on source resource.

Reply:
Yes, we still need to check those fields, I'm suggesting extracting `annotations` and `labels` from `metadata`.

Reply:
ok got it. Instead of deleting fields from `metadata` I can extract only  `annotations` and `labels`.